contracts/modules/TransferManager/GTM/GeneralTransferManager.sol

```diff
@@ -134,34 +134,85 @@ contract GeneralTransferManager is GeneralTransferManagerStorage, TransferManage
      * @notice Default implementation of verifyTransfer used by SecurityToken
      * @param _from Address of the sender
      * @param _to Address of the receiver
+     * @param _data off-chain signed data
     */
     function verifyTransfer(
         address _from,
         address _to,
         uint256 /*_amount*/,
-        bytes memory /* _data */
+        bytes memory _data
     )
         public
         view
         returns(Result, bytes32)
     {
-        return _verifyTransfer(_from, _to);
+        if (_data.length > 32) {
+            address target;
+            uint256 nonce;
+            uint256 validFrom;
+            uint256 validTo;
+            bytes memory data;
+            (target, nonce, validFrom, validTo, data) = abi.decode(_data, (address, uint256, uint256, uint256, bytes));
+            if (target == address(this)) {
+                (
+                    bool success, 
+                    address[] memory investor,
+                    ,
+                    uint256[] memory canRecieveAfter,
+                    uint256[] memory expiryTime
+                ) = _processTransferSignatureView(data, validFrom, validTo, nonce);
+                if (success) {
+                    for (nonce = 0; nonce < investor.length; nonce++) {
+                        // Searching the _to token holder to get the required details for _verifyTransfer validation 
+                        if (investor[nonce] == _to)
```

Comment:
This logic doesn't quite match the `transferWithData` version as it is possible the signed data is relevant for the `from` address which wouldn't be picked up here I think.

Reply:
Do you think `from` address will come under into the check, I am expecting `from` does posses the balance it means It is already whitelisted with the ST?